src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptionhandlers/ErrorNotificationExceptionHandler.java

```diff
@@ -1,38 +1,43 @@
 package uk.gov.hmcts.reform.bulkscanprocessor.exceptionhandlers;
 
-import com.microsoft.azure.servicebus.IMessage;
 import uk.gov.hmcts.reform.bulkscanprocessor.client.ErrorNotificationException;
-import uk.gov.hmcts.reform.bulkscanprocessor.model.ErrorNotificationMessageWrapper;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.servicebus.MessageAutoCompletor;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
 
 public class ErrorNotificationExceptionHandler {
 
-    public ErrorNotificationExceptionHandler() {
-        // empty construct
+    private final MessageAutoCompletor autoCompletor;
+
+    public ErrorNotificationExceptionHandler(MessageAutoCompletor autoCompletor) {
+        this.autoCompletor = autoCompletor;
     }
 
-    public ErrorNotificationMessageWrapper handle(IMessage message, Throwable throwable) {
+    public CompletableFuture<Void> handle(UUID lockToken, Throwable throwable) {
         // return for message session completion step
         return throwable != null
-            ? handleNonNullThrowable(message, throwable)
-            : ErrorNotificationMessageWrapper.forAcknowledgement(message);
+            ? handleNonNullThrowable(lockToken, throwable)
+            : autoCompletor.completeAsync(lockToken);
     }
 
-    private ErrorNotificationMessageWrapper handleNonNullThrowable(IMessage message, Throwable throwable) {
+    private CompletableFuture<Void> handleNonNullThrowable(UUID lockToken, Throwable throwable) {
         if (throwable instanceof ErrorNotificationException) {
-            return handleErrorNotificationException(message, (ErrorNotificationException) throwable);
+            return handleErrorNotificationException(lockToken, (ErrorNotificationException) throwable);
         } else {
-            return ErrorNotificationMessageWrapper.forDeadLettering(message);
+            return autoCompletor.deadLetterAsync(lockToken, "Unknown exception", throwable.getMessage());
         }
     }
 
-    private ErrorNotificationMessageWrapper handleErrorNotificationException(
-        IMessage message,
+    private CompletableFuture<Void> handleErrorNotificationException(
+        UUID lockToken,
         ErrorNotificationException exception
     ) {
         if (exception.getStatus().is5xxServerError()) {
```

Comment:
This condition is not a part of this PR, so nothing to change as a part of it, but here's a thought: what if we get a 408 (timeout), for example?

Reply:
Yes, I think I mentioned before that conditions to decide when to void/deadletter can be grained more finely later on